### PR TITLE
Add integration testing

### PR DESCRIPTION
**Pull request by daninfpj**
This PR adds integrations testing  (#2). It loads the example app as a submodule and runs the tests on top of that.

The example app was modified to remove CSS and JS assets because they were causing trouble when running the tests.

There's [a couple steps](https://github.com/daninfpj/webpacker-react/tree/testing#testing) that should be run before running the tests but I’m not sure if we should add them to the rake task.

There's also a couple additional commits:

- Rubocop was added to check style issues and the rules were copied over from rails
- The other one changes the helper we were using to render the div tag, because it was generating self-closing tags that eat the content that comes after them (similar to this [issue](https://github.com/rails/rails/issues/4919))

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+AllCops:
+  TargetRubyVersion: 2.2
+  # RuboCop has a bunch of cops enabled by default. This setting tells RuboCop
+  # to ignore them, so only the ones explicitly set in this file are enabled.
+  DisabledByDefault: true
+  Exclude:
+    - '**/templates/**/*'
+    - '**/vendor/**/*'
+    - 'actionpack/lib/action_dispatch/journey/parser.rb'
```
Comment (renchap) [inline .rubocop.yml]: This is probably not needed

File: test/rails_helper.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
-require "rails/all"
-require "action_view/testing/resolvers"
+# Configure Rails Environment
+ENV["RAILS_ENV"] = "test"
+
+require File.expand_path("../example_app/config/environment.rb", __FILE__)
 require "rails/test_help"
+require "rails/generators"
+require "pathname"
+require 'minitest/mock'
```
Comment (renchap) [inline test/rails_helper.rb]: Use double quotes for consistence

File: .rubocop.yml (hunk 1 above)
Comment (daninfpj) [inline .rubocop.yml]: Fixed. Left only `'**/vendor/**/*'`as some node_modules include ruby files.

File: test/rails_helper.rb (hunk 2 above)
Comment (daninfpj) [inline test/rails_helper.rb]: Fixed.

**Comment by daninfpj** [comment]
I added the submodule precisely because I wanted to avoid the duplication. In the end it was actually a bit cumbersome to work with it as you mention, so I removed it.

Also, I'm now using `yarn` everywhere.

File: test/example_app/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Webpacker-React sample application
```
Comment (renchap) [inline test/example_app/README.md]: Not needed

File: test/example_app/app/assets/config/manifest.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+//= link_tree ../images
```
Comment (renchap) [inline test/example_app/app/assets/config/manifest.js]: Not needed. Might be a good idea to remove the whole ´app/assets' directory if it does not break the app

File: test/example_app/app/helpers/pages_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module PagesHelper
```
Comment (renchap) [inline test/example_app/app/helpers/pages_helper.rb]: Same for helpers, lets try to keep the files down to a minimum

**Pull request merged**